LanguageExt.Core/Monads/Alternative Value Monads/Either/Either-Shared/EitherData.Extensions.cs

```diff
@@ -0,0 +1,64 @@
+﻿#nullable enable
+using System;
+using System.Threading.Tasks;
+using LanguageExt.Common;
+
+namespace LanguageExt.DataTypes.Serialisation
+{
+    public static class EitherDataExtensions
+    {
+        public static Either<L, R> ToEither<L, R>(this EitherData<L, R> input) =>
+            input.State switch
+            {
+                EitherStatus.IsBottom => Either<L, R>.Bottom,
+                EitherStatus.IsRight => Either<L, R>.Right(input.Right),
+                EitherStatus.IsLeft => Either<L, R>.Left(input.Left),
+                _ => throw new InvalidOperationException($"Invalid {nameof(EitherStatus)} with value: {input.State}"),
```

Comment:
Is this correct approach with `InvalidOperationException`, or to use something like

```suggestion
                EitherStatus.IsRight => Either<L, R>.Right(input.Right),
                EitherStatus.IsLeft => Either<L, R>.Left(input.Left),
                _ => Either<L, R>.Bottom,
```

Reply:
I think the exception approach is best here, as it is an exceptional and unknown value, defaulting to bottom is misleading I think.